### PR TITLE
Create EngineRunner to test engine functionality 

### PR DESCRIPTION
**Pull request by AminArria**
## Changes
The idea of `EngineRunner` is to eventually fully replace `Runner`, in addition it will contain the translation layer of LambdaGameEngine state into Myrra state as this translation will be crucial to not require big refactors in the client while allowing us to test the functionalities of LambdaGameEngine as soon as they are pushed 

Overall this PR introduces the following
- EngineRunner module
- Config to replace Runner with EngineRunner

It includes the following features in EngineRunner
-  Loot spawning 
-  Player movement (with loot pickup)
-  Player info (id, position, status, health, body_size, cooldowns)

PS: This PR needs https://github.com/lambdaclass/lambda_game_engine/pull/11
- [x] Update `mix.lock` once merged

## Testing this PR

In `config.exs` change the config to
```
config :dark_worlds_server, DarkWorldsServer.Engine.Runner,
  process_priority: :high,
  use_engine_runner: true  ## <- Setting true enables EngineRunner to replace Runner
```

### CODE REVIEW

File: server/lib/dark_worlds_server/engine/engine_runner.ex [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+    {game_state, _loot_id} = LambdaGameEngine.spawn_random_loot(state.game_state)
+
+    {:noreply, %{state | game_state: game_state}}
+  end
+
+  def handle_info(msg, state) do
+    Logger.error("Unexpected handle_info msg", %{msg: msg})
+    {:noreply, state}
+  end
```
Comment (agustinesco) [inline server/lib/dark_worlds_server/engine/engine_runner.ex:124]: Can we add this to the `handle_cast` too? while testing this pr i got the server crashing while trying to shoot with hack, i think it's worth handling

File: server/lib/dark_worlds_server/engine/engine_runner.ex [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    Enum.map(players, fn {_id, player} ->
+      %{
+        ## Transformed
+        __struct__: LambdaGameEngine.MyrraEngine.Player,
+        id: player.id,
+        position: transform_position_to_myrra_position(player.position),
+        status: if(player.health <= 0, do: :dead, else: :alive),
+        health: player.health,
+        body_size: player.size,
+        character_name: "H4ck",
```
Comment (agustinesco) [inline server/lib/dark_worlds_server/engine/engine_runner.ex:164]: This is a placeholder value while we implement the rest of the characters in the config file right?

File: server/lib/dark_worlds_server/engine/engine_runner.ex [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+  #######################
+  # GenServer callbacks #
+  #######################
+  @impl true
+  def init(%{engine_config_raw_json: engine_config_raw_json}) do
+    # priority =
+    #   Application.fetch_env!(:dark_worlds_server, __MODULE__)
+    #   |> Keyword.fetch!(:process_priority)
+
+    # Process.flag(:priority, priority)
```
Comment (agustinesco) [inline server/lib/dark_worlds_server/engine/engine_runner.ex:50]: Should we delete this lines? are they meant to be used in the future?

**Review by agustinesco — CHANGES_REQUESTED** [review]

File: server/lib/dark_worlds_server/engine/engine_runner.ex [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+      %{
+        id: loot.id,
+        loot_type: {:health, :placeholder},
+        position: transform_position_to_myrra_position(loot.position)
+      }
+    end)
+  end
+
+  defp transform_position_to_myrra_position(position) do
+    %LambdaGameEngine.MyrraEngine.Position{x: -1 * position.y + 5000, y: position.x + 5000}
```
[the remark this answers is not in the document]
Comment (AminArria) [reply server/lib/dark_worlds_server/engine/engine_runner.ex:206]: Yes, this is actually `height` and `width` divided by 2, so yeah I'll make the update

File: server/lib/dark_worlds_server/engine/engine_runner.ex (hunk 3 above)
Comment (AminArria) [reply server/lib/dark_worlds_server/engine/engine_runner.ex:50]: In theory we should have them to be like the other runner, I should probably uncomment it

File: server/lib/dark_worlds_server/engine/runner.ex [hunk 5]
```diff
@@ -577,4 +587,21 @@ defmodule DarkWorldsServer.Engine.Runner do
...
+    ## Start engine_runner
+    {:ok, engine_runner_pid} = Engine.start_engine_runner()
+
+    ## Setup each player in engine_runner
+    Enum.each(selected_players, fn {id, name} ->
+      :ok = EngineRunner.join(engine_runner_pid, id, String.downcase(name))
+    end)
+
+    ## Start the game ticks
+    EngineRunner.start_game_tick(engine_runner_pid)
```
[the remark this answers is not in the document]
Comment (AminArria) [reply server/lib/dark_worlds_server/engine/runner.ex:601]: The reason I didn't do it was to provide the abstraction to have that separate and in this particular case to have the chance to add all players before starting to send updates (the tick is the loop broadcasting the states)

File: server/lib/dark_worlds_server/engine/engine_runner.ex (hunk 4 above)
[the remark this answers is not in the document]
Comment (AminArria) [reply server/lib/dark_worlds_server/engine/engine_runner.ex]: Done in [0159682](https://github.com/lambdaclass/curse_of_myrra/pull/953/commits/01596825418f4ce98c933a154d2d5883c144d5f9)

File: server/lib/dark_worlds_server/engine/engine_runner.ex (hunk 1 above)
Comment (AminArria) [reply server/lib/dark_worlds_server/engine/engine_runner.ex:136]: Done in [0159682](https://github.com/lambdaclass/curse_of_myrra/pull/953/commits/01596825418f4ce98c933a154d2d5883c144d5f9)

File: server/lib/dark_worlds_server/engine/engine_runner.ex (hunk 2 above)
Comment (AminArria) [reply server/lib/dark_worlds_server/engine/engine_runner.ex]: Done in [0159682](https://github.com/lambdaclass/curse_of_myrra/pull/953/commits/01596825418f4ce98c933a154d2d5883c144d5f9)

File: server/lib/dark_worlds_server/engine/engine_runner.ex (hunk 3 above)
Comment (AminArria) [reply server/lib/dark_worlds_server/engine/engine_runner.ex]: Uncommented

File: server/lib/dark_worlds_server/engine/engine_runner.ex [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+      players: transform_players_to_myrra_players(game_state.players),
+      board: %{
+        width: game_state.config.game.width,
+        __struct__: LambdaGameEngine.MyrraEngine.Board,
+        height: game_state.config.game.height
+      },
+      projectiles: [],
+      killfeed: [],
+      playable_radius: 20_000,
+      shrinking_center: %LambdaGameEngine.MyrraEngine.Position{x: 5000, y: 5000},
```
[the remark this answers is not in the document]
Comment (AminArria) [reply server/lib/dark_worlds_server/engine/engine_runner.ex:157]: This is a placeholder, map shrinking is not implemented (yet)

File: server/lib/dark_worlds_server/engine/engine_runner.ex (hunk 1 above)
Comment (agustinesco) [reply server/lib/dark_worlds_server/engine/engine_runner.ex:136]: nice!

File: server/lib/dark_worlds_server/engine/runner.ex (hunk 5 above)
[the remark this answers is not in the document]
Comment (agustinesco) [reply server/lib/dark_worlds_server/engine/runner.ex:613]: oh nice! thanks for explaining it

File: server/lib/dark_worlds_server/engine/engine_runner.ex (hunk 4 above)
[the remark this answers is not in the document]
Comment (agustinesco) [reply server/lib/dark_worlds_server/engine/engine_runner.ex]: It's much clearer now! nice

**Review by agustinesco — APPROVED** [review]
LGTM!

**Review by manucamejo — APPROVED** [review]

**Pull request merged**